pages/feedingtime.js

```diff
@@ -0,0 +1,97 @@
+import { useSession, signIn } from "next-auth/react";
+import { unstable_getServerSession } from "next-auth/next";
+import { authOptions } from "./api/auth/[...nextauth]";
+import styled from "styled-components";
+import { CardContainer } from "../components/CardContainer";
+import Card from "../components/Card";
+import { getAllDays } from "../services/dayService";
+import { SignInButton } from "../components/SignInButton";
+
+import { CanvasContainer } from "../components/CanvasContainer";
+import LineChart from "../components/charts/LineChart";
+
+export async function getServerSideProps(context) {
+  const session = await unstable_getServerSession(
+    context.req,
+    context.res,
+    authOptions
+  );
+  if (session) {
+    const days = await getAllDays(session.user.email);
+    return {
+      props: { days: JSON.parse(JSON.stringify(days)) },
+    };
+  } else return { props: {} };
+}
+
+export default function FeedingTimePage({ days }) {
+  //----- Session -----
+  const { data: session } = useSession();
+
+  if (session) {
+    const filteredDays = days.filter((day) =>
+      Boolean(day.feastTimes.length > 0)
+    );
+
+    const ascendingFilteredDays = Array.from(filteredDays).reverse();
```

Comment:
Just as a side note: reversing an array is not the same as sorting it in ascending/descending order.
If reversing the array works for your use case, you're good. If however your intention is to actually sort the array in asc/desc order you need to use `sort`.

Reply:
you are right of course. it only works in this case because i already sort the array while fetching. so it already is sorted (for my cards - latest first) and then i need the reverse order (for my chart - latest first)

Reply:
That's exactly my point. If your array is already sorted, say in ascending order, and you then want to have it sorted in descending order, using `reverse` potentially gives you the wrong result. In your case, `reverse` may perform just as `sort`, but that's not true for all cases and I wanted to make you aware of it 😊